src/globus_sdk/transport/_clientinfo.py

```diff
@@ -124,3 +127,14 @@ def _check_reserved_chars(key: str, value: str) -> None:
             "X-Globus-Client-Info reserved characters cannot be used in keys or "
             f"values. Bad usage: '{key}: {value}'"
         )
+
+
+class RequestCallerInfo:
+    """
+    Data object that holds contextual information about the caller of a request.
+
+    :param authorizer: The authorizer object from the client making the request
+    """
+
+    def __init__(self, *, authorizer: GlobusAuthorizer | None = None) -> None:
+        self.authorizer = authorizer
```

Comment:
Let's put this somewhere else. Per the module docstring, this implements the `X-Globus-Client-Info` header spec. We shouldn't mix in other functionality.